synapse/handlers/sync.py

```diff
@@ -988,7 +988,9 @@ async def compute_state_delta(
             # sync's timeline and the start of the current sync's timeline.
             # See the docstring above for details.
             state_ids: StateMap[str]
-
+            # We need to know whether the state we fetch may be partial, so check
+            # whether the room is partial stated *before* fetching it.
+            is_partial_state_room = await self.store.is_partial_state_room(room_id)
```

Comment:
I guess to make this 100% robust we'd need to atomically fetch `is_partial_state_room` while we fetch the state ids?

(To be explicit, I don't think we should do this.)